04_numeronym/joshcarp/numeronym.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import (
+	"fmt"
+	"unicode"
+)
+
+func main() {
+	fmt.Println(numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+func numeronyms(vals ...string) []string {
+	result := make([]string, 0, len(vals))
+	for _, word := range vals {
+		word = removeWhitespaces(word)
+		if length := len(word); length > 3 && isWord(word) {
+			word = fmt.Sprintf("%s%d%s", string(word[0]), length-2, string(word[length-1]))
+		}
+		result = append(result, word)
+	}
+	return result
+}
+
+func removeWhitespaces(word string) string {
+	var newWord string
+	for _, letter := range word {
+		if letter != ' ' {
+			newWord += string(letter)
+		}
+	}
+	return newWord
+}
+func isWord(word string) bool {
```

Comment:
Cheeky - there are words created with letters, where the letters are still not ascii (use more than one byte and are not reflected in as expected in len(str) which returns bytes and number of characters), e.g. write testcases with German umlaut characters or similar.

You will need to convert your string to []rune slice and count runes (there are a bunch of existing helpers for this).